docs/apache-airflow/start/docker.rst

```diff
@@ -0,0 +1,170 @@
+2. Install `Docker Compose <https://docs.docker.com/compose/install/>`__ v1.27.0 and newer on your workstation.
+
+Older versions of ``docker-compose`` do not support all features required by ``docker-compose.yaml`` file, so double check that it meets the minimum version requirements.
+
+``docker-compose.yaml``
+=======================
+
+To deploy Airflow on Docker Compose, you should fetch `docker-compose.yaml <../docker-compose.yaml>`__.
+
+.. jinja:: quick_start_ctx
+
+    .. code-block:: bash
+
+        curl -LfO '{{ doc_root_url }}docker-compose.yaml'
+
+This file contains several service definitions:
+
+- ``airflow-scheduler`` - The :doc:`scheduler </scheduler>` monitors all tasks and DAGs, then triggers the
+  task instances once their dependencies are complete.
+- ``airflow-webserver`` - The webserver available at ``http://localhost:8080``.
+- ``airflow-worker`` - The worker that executes the tasks given by the scheduler.
+- ``airflow-init`` - The initialization service.
+- ``flower`` - `The flower app <https://flower.readthedocs.io/en/latest/>`__ for monitoring the environment. It is available at ``http://localhost:8080``.
+- ``postgres`` - The database.
+- ``redis`` - `The redis <https://redis.io/>`__ - broker that forwards messages from scheduler to worker.
+
+All these services allow you to run Airflow with :doc:`CeleryExecutor </executor/celery>`. For more information, see :ref:`architecture`.
+
+Some directories in the container are mounted, which means that their contents are synchronized between your computer and the container.
+
+- ``./dags`` - you can put your DAG files here.
+- ``./logs`` - contains logs from task execution and scheduler.
+- ``./plugins`` - you can put your :doc:`custom plugins </plugins>` here.
+
+Initializing Environment
+========================
+
+Before starting Airflow for the first time, You need to prepare your environment, i.e. create the necessary files, directories and initialize the database.
+
+On **Linux**, the mounted volumes in container use the native Linux filesystem user/group permissions, so you have to make sure the container and host computer have matching file permissions.
+
+.. code-block:: bash
+
+    mkdir ./dags ./logs ./plugins
+    echo -e "AIRFLOW_UID=$(id -u)\nAIRFLOW_GID=0" > .env
+
+On **all operating system**, you need to run database migrations and create the first user account. To do it, run.
+
+.. code-block:: bash
+
+    docker-compose up --rm airflow-init
+
+After initialization is complete, you should see a message like below.
+
+.. code-block:: bash
+
+    airflow-init_1       | Upgrades done
+    airflow-init_1       | Admin user airflow created
+    airflow-init_1       | 2.1.0
+    start_airflow-init_1 exited with code 0
+
+The account created has the login ``airflow`` and the password ``airflow``.
+
+Running Airflow
+===============
+
+Now you can start all services:
+
+.. code-block:: bash
+
+    docker-compose up
```

Comment:
In case of one customer we pause all DAGs before starting scheduler when running locally. This is a good idea if you want to run Airflow instance locally with production DAGs which often have schedules and may not work locally due to external dependencies.

Reply:
https://github.com/apache/airflow/pull/13660/files#diff-d6b3b45a036b2fb62762c6f8bc11f4b7832b3d012c81819dec51dc88ae304d54R50

Reply:
Isn't it overwritten by information from DAG?

Reply:
Yes. This can be overridden by a parameter in the DAG constructor.. This sounds like expected behavior. The user has control over how the environment behaves.